cache/tags_stats.go

```diff
@@ -0,0 +1,116 @@
+package cache
+
+import (
+	"database/sql"
+
+	"github.com/Bnei-Baruch/sqlboiler/queries"
+	"github.com/pkg/errors"
+	"gopkg.in/volatiletech/null.v6"
+
+	"github.com/Bnei-Baruch/archive-backend/mdb"
+)
+
+type TagsStatsCache interface {
+	Refresh() error
+	GetChildren(rootUIDs []string) ([]string, []int64)
+	GetHistogram() ClassByTypeStats
+}
+
+type TagsStatsCacheImpl struct {
+	mdb  *sql.DB
+	tree *StatsTree
+}
+
+func NewTagsStatsCacheImpl(mdb *sql.DB) TagsStatsCache {
+	stats := new(TagsStatsCacheImpl)
+	stats.mdb = mdb
+	return stats
+}
+
+func (s *TagsStatsCacheImpl) Refresh() error {
+	err := s.load()
+	return errors.Wrap(err, "Load tags and sources stats.")
+}
+
+func (s *TagsStatsCacheImpl) GetHistogram() ClassByTypeStats {
+	return s.tree.flatten()
+}
+
+func (s *TagsStatsCacheImpl) GetChildren(rootUIDs []string) ([]string, []int64) {
```

Comment:
Can we DRY up `GetChildren`, `getAllChildren` and `GetHistogram` ?
Maybe methods of `StatsTree` ?

Reply:
I dont know what data will need for souces and what will need for tags. Was think to do it after replace all places where use souce o tag tree from DB

Reply:
moved